gnosis/safe-react
# Feature: deep-linking a single tx

## What it solves
Resolves #970

## How this PR fixes it
A new route has been added `/{{shortName}}:{{safeAddress}}/transactions/{{txId}}`. A new component `<TxSingularDetails>` fetches the matching transaction and adds it to the store as the transaction components/hooks are deeply integrated with Redux.

The CGW returned transaction 'details' are converted to a 'summary' by means of a new function `makeTxFromDetails()` because the frontend `Transaction` type require a mixture of `TransactionSummary` and `TransactionDetails` from the CGW. The latter [can be used](https://gnosisinc.slack.com/archives/CQB729LLF/p1636448016168300?thread_ts=1636446393.164700&cid=CQB729LLF) to create a `Transaction`.

### Key changes:
- Copy deeplink button added to individual transactions
- Link has transaction id
- Deeplink page only has transaction details
- Tabs do not exist on deeplinked transactions because the title is self-explanatory
- "Details" breadcrumb
- Created transactions navigate to the deeplinked transaction
- Anonymised tracking event `.../transactions/TRANSACTION_ID`

### Other notable changes:
- The `gatewayTransactions` reducer was cleaned up.
- `txLocation` was removed from selectors.
- A new prop `iconType` was also added to SRC's copy button component, so that we can assign a custom icon to it.

## Known bugs:
The share button is disabled because of the following, which will be split into further issues.

- Sole-owned Safes can reject transaction a second time until next transaction list poll
- Multi-owned Safes must confirm rejections from initially queued transaction

## Required in next version:
- Grouped transactions (should fix multi-owned issue above)

## How to test it
1. Open the either the queued/historical transaction list(s) and expand a transaction.
2. Copy the new transaction link via the new share button.
3. Open the link and expect the transaction to be displayed in an expanded accordion, as it would be normally.

## Screenshots
![image](https://user-images.githubusercontent.com/20442784/141451145-4170c917-5031-46a8-8605-c2d80faa4fa7.png)
![image](https://user-images.githubusercontent.com/20442784/141451210-9d960db0-26a7-4790-8544-d4588e54a2df.png)
![image](https://user-images.githubusercontent.com/20442784/141451235-9da9e2ed-e20b-4fad-850b-c9a168ff3ff2.png)
![image](https://user-images.githubusercontent.com/20442784/141451265-358fc453-d53d-4f01-b3a5-8f40b5d5c199.png)
![image](https://user-images.githubusercontent.com/20442784/141451296-c664b98e-283a-4258-a7e9-48f70ed26c68.png)
